test/test_endpoints.py

```diff
@@ -350,3 +381,147 @@ def test_get_transaction_trace_of_rejected():
+            "entry_point_selector": "0xC73F681176FC7B3F9693986FD7B14581E8D540519E27400E88B8713932BE01",
+            "l1_contract_address": "0xe7f1725E7734CE288F8367e1Bb143E90bb3F0512",
+            "payload": ["0x1", "0x1"],
+            "nonce": "0x0",
+        }
+    )
+
+    assert response.status_code == 200
+    assert_tx_status(hex(response.json().get("invoke_tx_hash")), "REJECTED")
+
+
+@devnet_in_background(*PREDEPLOY_ACCOUNT_CLI_ARGS)
+def test_send_message_to_l1_deploy_execute():
+    """Test POST l2 to l1 deploy contract and execute transaction"""
+    deploy_info = deploy(L1L2_CONTRACT_PATH)
+
+    # increase and withdraw balance
+    invoke(
+        calls=[(deploy_info["address"], "increase_balance", [USER_ID, 3333])],
+        account_address=PREDEPLOYED_ACCOUNT_ADDRESS,
+        private_key=PREDEPLOYED_ACCOUNT_PRIVATE_KEY,
+    )
+    invoke(
+        calls=[
+            (
+                deploy_info["address"],
+                "withdraw",
+                [USER_ID, 1000, 0xE7F1725E7734CE288F8367E1BB143E90BB3F0512],
+            )
+        ],
+        account_address=PREDEPLOYED_ACCOUNT_ADDRESS,
+        private_key=PREDEPLOYED_ACCOUNT_PRIVATE_KEY,
+    )
+
+    response = send_message_to_l1(
+        {
+            "l2_contract_address": deploy_info["address"],
+            "l1_contract_address": "0xe7f1725E7734CE288F8367e1Bb143E90bb3F0512",
+            "payload": ["0x0", "0x1", "0x3e8"],
+        }
+    )
+
+    assert response.status_code == 200
+
+
+@devnet_in_background(*PREDEPLOY_ACCOUNT_CLI_ARGS)
+def test_send_message_to_l1_deploy_execute_without_withdraw():
+    """Test POST l2 to l1 deploy contract and try to execute transaction without calling withdraw"""
+    deploy_info = deploy(L1L2_CONTRACT_PATH)
+    response = send_message_to_l1(
+        {
+            "l2_contract_address": deploy_info["address"],
+            "l1_contract_address": "0xe7f1725E7734CE288F8367e1Bb143E90bb3F0512",
+            "payload": ["0x0", "0x1", "0x3e8"],
+        }
+    )
+
+    assert response.status_code == 500
+
+
+@devnet_in_background()
+def test_send_message_to_l1_execute_without_data():
+    """Test POST l2 to l1 deploy without data"""
+    response = send_message_to_l1(
+        {
+            "l2_contract_address": "",
+            "l1_contract_address": "",
+            "payload": "",
+        }
+    )
+
+    assert response.status_code == 400
+    assert response.json().get("code") == str(StarkErrorCode.MALFORMED_REQUEST)
+
+
+@devnet_in_background()
+def test_send_message_to_l1_execute_without_deploy():
+    """Test POST l2 to l1 without contract deploy"""
+    response = send_message_to_l1(
+        {
+            "l2_contract_address": "0x00285ddb7e5c777b310d806b9b2a0f7c7ba0a41f12b420219209d97a3b7f25b2",
+            "l1_contract_address": "0xe7f1725E7734CE288F8367e1Bb143E90bb3F0512",
+            "payload": ["0x0", "0x1", "0x3e8"],
+        }
+    )
+
+    assert response.status_code == 500
```

Comment:
Also assert StarkErrorCode as we already do

Reply:
https://github.com/Shard-Labs/starknet-devnet/pull/365/commits/5075e280efd28e04a85054785c9cd48ecc7fbe16